### ISSUE TITLE
Move `interface.utils.eval_results` to `interface(.base)`

### ISSUE THREAD

**Issue by mih** [datalad/datalad#6694]
There is pretty much no `Interface` implementation without it, just like with `build_doc`. IMHO both pieces should live in the same place.

This would make the common

```py
from datalad.interface.base import (
    Interface,
    build_doc,
)
from datalad.interface.utils import (
    eval_results,
)
```

just be 

```py
from datalad.interface.base import (
    Interface,
    build_doc,
    eval_results,
)
```

an argument could be made to consolidate all (more or less strictly) required pieces in `datalad.interface`. Looking like this:

```py
from datalad.interface.base import (
    Interface,
    Parameter,
    build_doc,
    datasetmethod,
    eval_results,
)
```

**Comment by adswa** [comment]
done in  #7170

**Comment by yarikoptic-gitmate** [comment]
Issue fixed in [`0.18.0`](https://github.com/datalad/datalad/releases/tag/0.18.0)
